packages/cli/src/Generate.ts

```diff
@@ -136,6 +138,12 @@ ${bold('Examples')}
         cliVersion: pkg.version,
         generatorNames: args['--generator'],
         postinstall: Boolean(args['--postinstall']),
+        noEngine:
+          Boolean(args['--no-engine']) ||
+          Boolean(args['--data-proxy']) || // legacy
+          Boolean(args['--accelerate']) || // legacy
+          Boolean(process.env.PRISMA_GENERATE_DATAPROXY) || // legacy
+          Boolean(process.env.PRISMA_GENERATE_ACCELERATE), // legacy
```

Comment:
```suggestion
          Boolean(args['--data-proxy']) || // legacy, we need to keep it to avoid an uneeded breaking change
          Boolean(args['--accelerate']) || // legacy, we need to keep it to avoid an uneeded breaking change
          Boolean(process.env.PRISMA_GENERATE_DATAPROXY) || // legacy, we need to keep it to avoid an uneeded breaking change
          Boolean(process.env.PRISMA_GENERATE_ACCELERATE), // legacy, we need to keep it to avoid an uneeded breaking change
```